components/security-mgt/org.wso2.carbon.security.mgt/src/test/java/org/wso2/carbon/security/keystore/KeyStoreAdminTest.java

```diff
@@ -159,24 +138,53 @@ public void testGetTrustStoreSuccess() throws Exception {
     @Test(description = "Test case to verify that an exception is thrown when getting TrustStore")
     public void testGetTrustStoreException() throws Exception {
 
-        keyStoreAdmin = new KeyStoreAdmin(tenantID, registry);
-
         // Mock ServerConfiguration static calls
-        try (MockedStatic<ServerConfiguration> serverConfiguration = mockStatic(ServerConfiguration.class)) {
+        try (MockedStatic<KeyStoreManager> keyStoreManager = mockStatic(KeyStoreManager.class)) {
 
-            // Mocking configuration properties
-            serverConfiguration.when(ServerConfiguration::getInstance).thenReturn(this.serverConfiguration);
-            when(this.serverConfiguration.getFirstProperty(SERVER_TRUSTSTORE_FILE)).thenReturn(createPath(KEYSTORE_NAME).toString());
-            when(this.serverConfiguration.getFirstProperty(KeyStoreAdmin.SERVER_TRUSTSTORE_TYPE)).thenReturn("PKCS12");
-            when(this.serverConfiguration.getFirstProperty("JCEProvider")).thenReturn("BC");
+            keyStoreManager.when(() -> KeyStoreManager.getInstance(anyInt())).thenReturn(this.keyStoreManager);
+            when(this.keyStoreManager.getTrustStore())
+                    .thenThrow(new CarbonException("Error occurred while retrieving TrustStore"));
 
+            keyStoreAdmin = new KeyStoreAdmin(tenantID, registry);
             // Execute method under test
             assertThrows(SecurityConfigException.class, () -> {
                 keyStoreAdmin.getTrustStore();
             });
         }
     }
 
+    @Test(description = "Delete KeyStore test", dependsOnMethods = "testAddKeyStore")
+    public void testDeleteKeyStore() throws Exception {
+
+        try (MockedStatic<KeyStoreManager> keyStoreManager = mockStatic(KeyStoreManager.class);
+             MockedStatic<KeyStoreUtil> keyStoreUtil = mockStatic(KeyStoreUtil.class)) {
+
+            keyStoreManager.when(() -> KeyStoreManager.getInstance(anyInt())).thenReturn(this.keyStoreManager);
+
+            keyStoreUtil.when(() -> KeyStoreUtil.isPrimaryStore(anyString())).thenReturn(false);
+            keyStoreUtil.when(() -> KeyStoreUtil.isTrustStore(anyString())).thenReturn(false);
+
+            keyStoreAdmin = new KeyStoreAdmin(tenantID, registry);
+            keyStoreAdmin.deleteStore("new_keystore.jks");
```

Comment:
What is asserted here ?

Reply:
Let's use this `verify(keyStoreManager).deleteStore("new_keystore.jks");`